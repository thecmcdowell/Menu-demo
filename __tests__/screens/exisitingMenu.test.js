import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { render, cleanup } from 'react-native-testing-library';
import ExisitingMenu from '../../app/screens/existingMenu'

afterEach(cleanup)

jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual("@react-navigation/native")
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: jest.fn
        })
    }
})

describe('Exisiting Menu screen tests', () => {
    const mockStore = configureStore([])

    it('should display current menu items', () => {
        const store = mockStore({
            menu: [
                {
                    id: 1,
                    image: 'https://www.listchallenges.com/f/items/4a98113a-62e5-444a-82a1-627089b81bbb.jpg',
                    title: 'Taco',
                    description: 'its NOT sushi?',
                    price: '$5.99'
                }
            ]
        })

        const rendered = render(
            <Provider store={store}>
                <ExisitingMenu />
            </Provider>
        )

        expect(rendered).toMatchSnapshot()
    })
    it('shows there is no exisiting items', () => {
        const store = mockStore({ menu: [] })
        const rendered = render(
            <Provider store={store}>
                <ExisitingMenu />
            </Provider>
        )

        expect(rendered).toMatchSnapshot()
    })
})