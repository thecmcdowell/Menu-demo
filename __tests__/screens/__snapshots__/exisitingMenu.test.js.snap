// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exisiting Menu screen tests should display current menu items 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add Menu Item
      </Text>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "description": "its NOT sushi?",
          "id": 1,
          "image": "https://www.listchallenges.com/f/items/4a98113a-62e5-444a-82a1-627089b81bbb.jpg",
          "price": "$5.99",
          "title": "Taco",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "width": "100%",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 15,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "margin": 5,
                "padding": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://www.listchallenges.com/f/items/4a98113a-62e5-444a-82a1-627089b81bbb.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 150,
                  }
                }
              />
              <Text>
                Taco
              </Text>
            </View>
            <View
              style={
                Object {
                  "padding": 25,
                }
              }
            >
              <Text>
                its NOT sushi?
              </Text>
              <Text>
                $
                $5.99
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Exisiting Menu screen tests shows there is no exisiting items 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add Menu Item
      </Text>
    </View>
  </View>
  <Text>
    you have no menu items. Add some? 
  </Text>
</View>
`;
